### ISSUE TITLE
[] operator not supported for strings

### ISSUE THREAD

**Issue by Spuds** [elkarte/Elkarte#2946]
PHP 7.1 ... in function detectFulltextIndex

Seems doing 
```
$a = ''
$a[] = 'bla'
```
is fatal in 7.1 ...  so change ```$context['fulltext_index'] = '';``` to ```$context['fulltext_index'] = array();``` and ``` if (is_array($context['fulltext_index']))``` to ```if (!empty($context['fulltext_index']))```  

I do not immediaty see where it being a string (vs empty array) iis required, although it is in a global

**Comment by emanuele45** [comment]
If I read the code correctly, the array-form is used only in alterFullTextIndex to drop the indexes.
Also the global seems to be used in the template only to check for !empty, so if we want to break everything we could:
https://github.com/elkarte/Elkarte/commit/90442c0e40cd26f6254e942768eb58ea97d6b040

**Comment by Spuds** [comment]
Seems reasonable ...  if the only check is already !empty then '' or array() will be the same.

This is probably in 1.1 as well, have not checked.

**Comment by emanuele45** [comment]
Yep, it's 1.1 as well, in fact the commit over there is for 1.1. :angel:

**Comment by emanuele45** [comment]
Fixed in 1.1 with 9850920fbc900cecc6df9e9c270ea42e7b3c1b7b to backport to 1.0.x
